la_funding_analysis/pipeline/cleaning_and_joining.py

```diff
@@ -0,0 +1,224 @@
+# File: pipeline/cleaning_and_joining.py
+"""Cleans and joins the imported datasets to form
+a dataframe info_sp containing all information about
+local authorities and the grants they received.
+"""
+
+import numpy as np
+import pandas as pd
+from titlecase import titlecase
+
+from la_funding_analysis.getters.local_authority_data import (
+    get_EPC_means,
+    get_grants,
+    get_imd,
+    get_parties_models,
+    get_fuel_poverty,
+    get_social_potential,
+)
+from la_funding_analysis.utils.name_cleaners import clean_names, model_type
+
+
+def form_full_dataset() -> pd.DataFrame:
```

Comment:
Generally we haven't been using type hints up until now (although this is up for discussion). Either way it's best not to mix within a project

Reply:
This is a really big function! I would consider how you could break it down into smaller chunks that perform logical steps and then use a wrapper function to combine them.

Reply:
I've removed the type hint and broken this down into a sequence of cleaning/joining functions